lib/Ride_History.dart

```diff
@@ -22,82 +20,40 @@ class _RideHistoryState extends State<RideHistory> {
-        letterSpacing: 0.2,
-        fontSize: 22,
-        height: 2);
-
+  Widget _mainHist(List<Ride> rides) {
+    List<Widget> rideCards = [];
+    for (int i = 0; i < rides.length; i++) {
+      if (i == 0) {
+        rideCards.add(SizedBox(width: 16));
+      }
+      rideCards.add(Container(
+        width: MediaQuery.of(context).size.width * 0.65,
+        child: RideCard(rides[i],
+            showConfirmation: false, showCallDriver: false, showArrow: false),
+      ));
+      rideCards.add(SizedBox(width: 16));
+    }
     return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
       children: <Widget>[
-        ListView.builder(
-          shrinkWrap: true,
-          itemCount: rides.length,
-          itemBuilder: (c, int index) => RideHistoryCard(
-            timeDateWidget: Padding(
-              padding: const EdgeInsets.only(left: 20.0),
-              child: RichText(
-                text: TextSpan(
-                    text:
-                        DateFormat('MMM').format(rides[index].startTime) + ' ',
-                    style: monthStyle,
-                    children: [
-                      TextSpan(
-                          text: humanize.ordinal(int.parse(DateFormat('d')
-                                  .format(rides[index].startTime))) +
-                              ' ',
-                          style: dayStyle),
-                      TextSpan(
-                          text: DateFormat('jm').format(rides[index].startTime),
-                          style: timeStyle)
-                    ]),
-              ),
-            ),
-            infoRowWidget: InformationRow(
-                start: rides[index].startLocation,
-                end: rides[index].endLocation),
-          ),
-        )
+        SingleChildScrollView(
+          scrollDirection: Axis.horizontal,
+          child: Row(children: rideCards),
+        ),
       ],
     );
   }
 
   @override
-  Widget build(context) {
+  Widget build(BuildContext context) {
     RidesProvider ridesProvider = Provider.of<RidesProvider>(context);
-    AuthProvider authProvider = Provider.of(context);
-    AppConfig appConfig = AppConfig.of(context);
-
-    return FutureBuilder<List<Ride>>(
-        future: ridesProvider.fetchPastRides(context, appConfig, authProvider),
-        builder: (context, snapshot) {
-          if (snapshot.hasData) {
-            if (snapshot.data.length == 0) {
-              return _emptyRideHist(context);
-            } else {
-              return _mainHist(context, snapshot.data);
-            }
-          } else if (snapshot.hasError) {
-            return Text("${snapshot.error}");
-          }
-          return Center(child: CircularProgressIndicator());
-        });
+    List<Ride> pastRides = ridesProvider.pastRides;
+    if (pastRides.length == 0) {
+      return _emptyRideHist();
+    } else {
+      return _mainHist(pastRides);
```

Comment:
Is there a limit to how many past rides in Ride History we display on the home page?

Reply:
Since Aiden's gone we're just gonna pick 5 since that sounds reasonable--just added this, thanks for the catch!